elasticsearch/probe/readiness.sh

```diff
@@ -19,20 +19,45 @@
 # TODO: try re-use code from ./run.sh
 ES_REST_BASEURL=https://localhost:9200
 EXPECTED_RESPONSE_CODE=200
+EXPECTED_SG_DOCUMENT_COUNT=5
 secret_dir=/etc/elasticsearch/secret
 max_time=${READINESS_PROBE_TIMEOUT:-30}
 
-response_code=$(curl -s -X HEAD \
-    --cacert $secret_dir/admin-ca \
-    --cert $secret_dir/admin-cert \
-    --key  $secret_dir/admin-key \
-    --max-time $max_time \
-    -w '%{response_code}' \
-    "${ES_REST_BASEURL}/")
 
-if [ ${response_code} == ${EXPECTED_RESPONSE_CODE} ]; then
-  exit 0
-else
-  echo "Elasticsearch node is not ready to accept HTTP requests yet [response code: ${response_code}]"
+function check_if_ready() {
+  path="$1"
+  err_msg="$2"
+  response_code=$(curl -s --head \
+      --cacert $secret_dir/admin-ca \
+      --cert $secret_dir/admin-cert \
+      --key  $secret_dir/admin-key \
+      --max-time $max_time \
+      -o /dev/null \
+      -w '%{response_code}' \
+      "${ES_REST_BASEURL}${path}")
+
+  if [ "${response_code}" != ${EXPECTED_RESPONSE_CODE} ]; then
+    echo "${err_msg} [response code: ${response_code}]"
```

Comment:
- not sure what are the guidelines about readiness probes printing messages - I don't know what will happen, but I don't want to spew hundreds of error messages at the user about es not being available if it is a normal condition during startup
- if we do want these error messages, they should be more descriptive - the above message is meaningless to a user, and they don't know what else to check, or what actions to take to correct the situation

Reply:
> not sure what are the guidelines about readiness probes printing messages - I don't know what will happen, but I don't want to spew hundreds of error messages at the user about es not being available if it is a normal condition during startup

maximal number of error messages this could produce is 9 (but it is usually around 0-3, given the probe initial delay is 10s and period is 5s)
- ES not available at all
- search guard index not available
- search guard document count not as expected (5x)
- index template missing (2x)

And wouldn't it be useful to know what went awry when the pod doesn't start due to the readiness probe? Or should it rather be logged elsewhere?

Reply:
Yes, it would be useful to know.  Ideally, the user would only see an error message for the _last_ failed probe, instead of for _all_ of them.

Reply:
That is exactly what it does per readiness probe instance